Fixes to MigrateUserRolesToParticipatoryProcessRoles

danigb opened the pull request:
- Use organization_id field instead of the relation to check the conditions

#### :tophat: What? Why?

I've worked with @xredo to update the [decidim-diba](https://github.com/diputacioBCN/decidim-diba) to the latest gem version and we had to make some minimal changes to `MigrateUserRolesToParticipatoryProcessRoles` because it was failing.

This PR doesn't change the behaviour, it just changes the way it checks some conditions.

#### :pushpin: Related Issues

I didn't create a new issue, but I can do it if you prefer.

mrcasals commented:
Hi @danigb, thanks for your contribution and welcome to the project!

CodeClimate build is failing, can you fix it please?

danigb commented:
Hi @mrcasals,

Thanks for the welcome! I've appended a commit. Hope it works 👍

danigb commented:
Hi @deivid-rodriguez,

Thanks for reviewing. Yes, it seems that the code could be simplified. My idea was to change the little as possible. I can write a more concise implementation, but I don't know it's worth, just for some migrations.

deivid-rodriguez commented:
If it improves things, I think we can do it! :) But I agree it's not a big deal, so up to you!

mrcasals commented:
@danigb sorry again, can you add a CHANGELOG entry? In the `Fixed` section of the `UNRELEASED` section, please 😄

danigb commented:
Yes... I'll simplify the code with the suggestions of @deivid-rodriguez and add an entry to the CHANGELOG. 👍 Coming soon...

danigb commented:
Hi @mrcasals and @deivid-rodriguez 

I've refactored a little bit the code. I've extracted the query into a function and used `flat_map` instead of `map` to explicitly show that is a nested array (not strictly necessary since `join` flattens the array)

And of course the changelog entry 

Hope you like it

josepjaume commented:
Nice! Merging this!
